pep-3156.txt

```diff
@@ -1647,15 +1720,31 @@ callback-based frameworks like Twisted.  After converting a coroutine
 into a Task, callbacks can be added to the Task.
 
 To convert a coroutine into a task, call the coroutine function and
-pass the resulting coroutine object to the ``asyncio.Task()``
-constructor.  You may also use ``asyncio.async()`` for this purpose.
+pass the resulting coroutine object to the ``loop.create_task()``
+method.  You may also use ``asyncio.ensure_future()`` for this purpose.
 
 You may ask, why not automatically convert all coroutines to Tasks?
 The ``@asyncio.coroutine`` decorator could do this.  However, this would
 slow things down considerably in the case where one coroutine calls
 another (and so on), as switching to a "bare" coroutine has much less
 overhead than switching to a Task.
 
+A Task is inherited from ``Future`` but adds new methods:
+
+- ``current_task(loop=None)``. A *class method* returning the
```

Comment:
They might as well be static methods, and nobody would care. I'm fine with adding Task.set_current_task(); this is currently done by Task._step() reaching into the data structure used to hold these. (I'd propose to make set_current_task(None) delete it, that functionality is also needed.)

I agree with Andrew that moving these to the loop would be nice but shouldn't be done this close to the 3.6 release. I can't recall why we did it this way, probably because we didn't want to touch the (much more complicated) EventLoop code if we could solve this locally in Task. But with create_task() and the desire to separate coroutine runners from event loops that's perhaps now backwards. I guess adding set_current_task() makes everything at least possible.